### PR TITLE
fix(es/parser): export default/namespace/named

### LINKED ISSUE #4369: The parser does not throw an error for a namespace export with a named export
### Describe the bug

Named export is not supposed to be worked with export namespaced at the same time.
[Rollup Repl](https://rollupjs.org/repl/?version=2.70.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBhcyUyMG1haW4lMkMlMjAlN0IlMjBjdWJlJTIwJTdEJTIwZnJvbSUyMCU1QyUyMi4lMkZtYWluJTVDJTIyJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybWF0aHMuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwc3F1YXJlJTIwZnJvbSUyMCcuJTJGc3F1YXJlLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwJTdCZGVmYXVsdCUyMGFzJTIwc3F1YXJlJTdEJTIwZnJvbSUyMCcuJTJGc3F1YXJlLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBjdWJlJTIwKHglMjApJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBzcXVhcmUoeCklMjAqJTIweCUzQiU1Q24lN0QlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyc3F1YXJlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMHNxdWFyZSUyMCglMjB4JTIwKSUyMCU3QiU1Q24lNUN0cmV0dXJuJTIweCUyMColMjB4JTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

### Input code

```typescript
export * as foo, { bar } from "./foo"
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "target": "es2022",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "commonjs"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.157&code=H4sIAAAAAAAAA0utKMgvKlFISU3OSSxKVUgrzUsuyczPU0jLz7ex00i0UgjRtFIoy89MsebiSoWo1VJILAbJ6yhUK5SkFpco1CqkFeXnKijp6ScWp6QpAQCCetchUwAAAA%3D%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1FsXL3KGHMK4SHOIPGgdqjO9euThkJ2nePDU64GhtlK2AZUyosdgvrVRqZjjxudBCBXra7AnuuljZuSjCMC9jND5TAk9goeCj3%2BqQuRSyMPBENu7nTXZ1hfS5xqH9%2F6lTKyHFA9Qf0Sx7vCdd5OL%2BAwzZOlG9AAAA

### Expected behavior

Throw an syntax error.

### Actual behavior

Successfully parsed

### Version

1.2.157

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #4369

### CODE REVIEW

**Comment by magic-akari** [comment]
But personally, I hope this is a valid JavaScript statement.
Looking forward to reverting this change in the future.

**Comment by magic-akari** [comment]
Maybe I am wrong.
I found it's possible with `export default from`

>Note: The compound form must also support [export-ns-from](https://github.com/leebyron/ecmascript-export-ns-from) should both proposals be accepted:
>
>```JavaScript
>export v, * as ns from "mod";
>```

File: crates/swc_ecma_parser/src/parser/stmt.rs [hunk 1]
```diff
@@ -1644,6 +1644,67 @@ let x = 4";
...
+                ..Default::default()
+            }),
+            |p| p.parse_module(),
+        );
+    }
+
+    #[test]
+    fn issue_4369_4() {
+        test_parser(
+            r#"export * as bar, { baz } from "mod""#,
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/parser/stmt.rs:1687]: I think this should panic (with tested error message)?

Comment (magic-akari) [reply crates/swc_ecma_parser/src/parser/stmt.rs:1687]: Is there any spec of compound rules?
It's a little weird if we allow issue_4369_2, issue_4369_3, issue_4369_5 but disallow issue_4369_4.

I expects the default/namespace/name could be compounded arbitrarily if [export default from](https://github.com/tc39/proposal-export-default-from) proposal is accepted.

Comment (kdy1) [reply crates/swc_ecma_parser/src/parser/stmt.rs:1687]: I think this is not allowed at the moment, and I'm not sure if the proposal for `export default from` wil touch this one

File: crates/swc_ecma_transforms_proposal/tests/export.rs [hunk 2]
```diff
@@ -45,7 +45,7 @@ export { x, y as w } from "mod";
 );
 
 test!(
-    syntax_namespace(),
+    syntax_default(),
     |_| tr(),
     namespace_compound_es6,
     r"export * as foo, { bar } from 'bar';",
```
Comment (magic-akari) [inline crates/swc_ecma_transforms_proposal/tests/export.rs:51]: @kdy1 check this one. Is this wrong test case?

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_parser

**Pull request merged**